File: backend/src/main/java/org/backend/user/UserData.java
package org.backend.user;

import org.backend.misc.Util;

import java.io.Serializable;

public class UserData implements Serializable {
    public final int userId;
    public String username;
    public byte[] profilePicture;

    public UserData(int userId) {
        this.userId = userId;
    }

    public int getUserId() {
        return userId;
    }

    public String getUsername() {
        return username;
    }

    public byte[] getProfilePicture() {
        return profilePicture;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public void setProfilePicture(String profilePicture) {
        this.profilePicture = Util.imageToByteArray(profilePicture, 150, 150, 0.75f);
    }
}

File: frontend/app/src/main/java/com/example/distributed_hotel_booking/data/UserData.kt
package com.example.distributed_hotel_booking.data;

import java.util.Objects

class UserData(var userId: Int = 0, var username: String = "", var profilePicture: ByteArray = ByteArray(0)) {

override fun hashCode(): Int {
    val hash = Objects.hash(username)
    return hash and Int.MAX_VALUE
}

override fun toString(): String {
    return "UserData{" +
            "id=" + userId +
            ", username='" + username + '\'' +
            '}'
}
}
